plugins/inputs/t128_tank/reader.go

```diff
@@ -270,6 +276,7 @@ func (r *Reader) readFromTank(readCtx context.Context, startingIndex index) (err
 		var collectedMessages []IndexedMessage
 		for _, message := range messages {
 			collectedMessages = append(collectedMessages, *message)
+			r.lastSavedIndex = message.Index.value
```

Comment:
This is the point when the index should be conditionally saved. But you've made it thread unsafe because the value is being updated and accessed from different routines.

- [ ] make thread safe